tools: update eslint to latest

BridgeAR opened the pull request:
<!--
Thank you for your pull request. Please provide a description above and review
the requirements below.

Bug fixes and new features should include tests and possibly benchmarks.

Contributors guide: https://github.com/nodejs/node/blob/master/CONTRIBUTING.md
-->

##### Checklist
<!-- Remove items that do not apply. For completed items, change [ ] to [x]. -->

- [x] `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/doc/guides/contributing/pull-requests.md#commit-message-guidelines)

BridgeAR commented:
Lite CI https://ci.nodejs.org/job/node-test-pull-request-lite-pipeline/3674/

Trott commented:
Nit 1: Please specify what version you are upgrading to. (Preferably what version we're upgrading from as well.)

Nit 2: If I'm not mistaken, unneeded `disable` comments will result in errors with our configuration. If I'm right about that, then `make lint-js` will fail when the first commit lands but the second does not. If that's the case (and again, I believe it is), please squash the two commits into a single commit so that there isn't a commit where linting fails.

BridgeAR commented:
@Trott thanks, I forgot that it would complain about the comments if they are not required. I updated the PR and incorporated the suggestions.

Lite-CI https://ci.nodejs.org/job/node-test-pull-request-lite-pipeline/3675/

BridgeAR commented:
Rebased due to conflicts.

Lite-CI https://ci.nodejs.org/job/node-test-pull-request-lite-pipeline/3686/

BridgeAR commented:
New CI https://ci.nodejs.org/job/node-test-pull-request-lite-pipeline/3694/

ZYSzys commented:
@BridgeAR `eslint@6.0.0` was just released. How about update to v6.0.0 directly ?

Refs: https://github.com/eslint/eslint/releases/tag/v6.0.0

BridgeAR commented:
Updated to the latest release.

cjihrig commented:
ESLint 6.0.1 was released. Please skip 6.0.0 before landing this.

BridgeAR commented:
Lite CI https://ci.nodejs.org/job/node-test-pull-request-lite-pipeline/3721/

BridgeAR commented:
Landed in ed8fc7e11d688cbcdf33d0d149830064758bdcd2
